src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -389,13 +393,29 @@ public DType GetType(TexlNode node)
             return _typeMap[node.Id];
         }
 
+        public bool IsNodeValid(TexlNode node)
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

Nit: can we add a comment here explaining. 

// Ensure that this TexlNode is associated with this binding. 
// This is critical so that the node.Ids are valid.